### PR TITLE
ARROW-15165: [Python] Expose function to resolve S3 bucket region

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

**Comment by pitrou** [comment]
Based on PR #11997

**Comment by pitrou** [comment]
Also cc @amol-

File: python/pyarrow/_s3fs.pyx [hunk 1]
```diff
@@ -54,6 +54,31 @@ def finalize_s3():
     check_status(CFinalizeS3())
 
 
+def resolve_s3_region(bucket):
```
Comment (amol-) [inline python/pyarrow/_s3fs.pyx:57]: Should we add this to `docs/api/filesystems.rst` ?

Comment (pitrou) [reply python/pyarrow/_s3fs.pyx:57]: Good point, I'll do it.

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good, thanks!

**Pull request closed without merging**